SwiftCharts/Views/StraightLinePathGenerator.swift

```diff
@@ -30,7 +30,28 @@ open class StraightLinePathGenerator: ChartLinesViewPathGenerator {
                 progressline.addLine(to: p2)
             }
         }
-
+        
+        return progressline
+    }
+    
+    open func generateAreaPath(points: [CGPoint], lineWidth: CGFloat) -> UIBezierPath {
+        let progressline = UIBezierPath()
+        progressline.lineWidth = 1.0
+        progressline.lineCapStyle = .round
+        progressline.lineJoinStyle = .round
+        
+        if let p = points.first {
+            progressline.move(to: p)
+        }
+        
+        for i in 1..<points.count {
```

Comment:
This would crash if points is empty

Reply:
Good catch :) I just copied the existing path generation algorithm for the gradient area which was used to draw a gradient under a straight line graph.